### DISCUSSION TITLE
How can I wait for the callback?

### DISCUSSION THREAD

**Discussion by quanpham88** [tdlight-team/tdlight-java#106] [Q&A] [answered]
Is it possible to wait for callback and get the result of the function/request?
Tried to use CountDownLatch to wait but it does not seem to work, seem like only 1 task can be executed at a time so if I pause the main thread, callback is never executed.

**Comment by cavallium** [comment, marked as answer]
You should **never** block the tdlib thread. This is an asynchronous library.

**Reply by quanpham88** [reply]
Sorry I am not clearly understand what you say. Can you tell me if my understand is right or wrong
My understand is as below
1. One client (created using ClientManager) can only handle 1 task/callback handler at a time -> so if I want to send message to 20 users and putting 3 minutes between sending message (to avoid my account getting banned), then update message handler will not be called until all send messages finished?.

2. if 1 is right, is there anyway to make client.send in 1 thread, while its callback handler will be in different thread? I only need to put interval between api request sending, not the callback handler.

3. If I create multiple client using ClientManager, what will happen? Are multiple clients still subject to only 1 task at a time OR each client handle their task separately in different threads?
